py/miniconf-mqtt/miniconf/__main__.py

```diff
@@ -61,11 +63,21 @@ def main():
         prefix = devices.pop()
         logging.info('Automatically using detected device prefix: %s', prefix)
 
+    async def list_paths():
+        interface = await Miniconf.create(prefix, args.broker)
+        for path in await interface.list_paths():
+            value = await interface.get(path)
+            print(f'{path} = {value}')
+
+    if args.list:
+        loop.run_until_complete(list_paths())
+        return
```

Comment:
We could alternatively just place this as an activity to be completed immediately after all the `set()` calls complete (i.e. if none are supplied, just list settings), which would allow the user to view settings after application as well.

Reply:
I would be a bit hesitant to default to getting all paths as well. What about  behaving similar to e.g. Linux tools like sysctl:

```
rj@grouch:~/src/sinara-systems (master)$ sysctl kernel.pty
kernel.pty.max = 4096
kernel.pty.nr = 15
kernel.pty.reserve = 1024
rj@grouch:~/src/sinara-systems (master)$ sysctl kernel.pty.max
kernel.pty.max = 4096
rj@grouch:~/src/sinara-systems (master)$ sysctl kernel.pty.max=9
sysctl: permission denied on key "kernel.pty.max", ignoring
rj@grouch:~/src/sinara-systems (master)$ sysctl -N kernel.pty.max
kernel.pty.max
rj@grouch:~/src/sinara-systems (master)$ sysctl -h

Usage:
 sysctl [options] [variable[=value] ...]

Options:
  -a, --all            display all variables
  -A                   alias of -a
  -X                   alias of -a
      --deprecated     include deprecated parameters to listing
  -b, --binary         print value without new line
  -e, --ignore         ignore unknown variables errors
  -N, --names          print variable names without values
  -n, --values         print only values of the given variable(s)
  -p, --load[=<file>]  read values from file
  -f                   alias of -p
      --system         read values from all system directories
  -r, --pattern <expression>
                       select setting that match expression
  -q, --quiet          do not echo variable set
  -w, --write          enable writing a value to variable
  -o                   does nothing
  -x                   does nothing
  -d                   alias of -h

 -h, --help     display this help and exit
 -V, --version  output version information and exit

For more details see sysctl(8).
```